flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -233,10 +236,20 @@ public void run() {
 				while (this.isRunning) {
 
 					synchronized (checkpointLock) {
+
+						if (!this.isFormatOpen) {
+							this.format.openInputFormat();
+							this.isFormatOpen = true;
+						}
+
 						if (this.currentSplit != null) {
 
 							if (currentSplit.equals(EOS)) {
 								isRunning = false;
+								if (this.isFormatOpen) {
```

Comment:
is this necessary? Wouldn't we directly go into the `finally` clause because of the `break` where the IF is closed anyway?